### ISSUE TITLE
Will it be possible to load 70b models soon?

### ISSUE THREAD

**Issue by ogre76** [SciSharp/LLamaSharp#98]
Do you plan that it will be possible to load 70b models?

**Comment by martindevans** [comment]
I haven't tried it, but I believe 70B models should be supported on the 0.4.2 version at the moment.

**Comment by martindevans** [comment]
I did some more investigation into this to see what was required. Turns out the model I was testing with before was corrupt!

If you set `GroupedQueryAttention = 8` in the model params you can load llama2 70B right now 🥳

**Comment by ogre76** [comment]
How can i set the GroupedQueryAttention parameter in LLamaSharp? Thx

**Comment by martindevans** [comment]
I used the [q3_K_S](https://cdn-lfs.huggingface.co/repos/fe/d7/fed75e74ade1c82e6c2c6f5a570535c19702e32429288de1bc12737a73f73327/ad7b637390e3bd6b71d0103ece52967a7e73925f9672227b0cb7a2a90a42eb25?response-content-disposition=attachment%3B+filename*%3DUTF-8%27%27llama-2-70b.ggmlv3.q3_K_S.bin%3B+filename%3D%22llama-2-70b.ggmlv3.q3_K_S.bin%22%3B&response-content-type=application%2Foctet-stream&Expires=1692135828&Policy=eyJTdGF0ZW1lbnQiOlt7IkNvbmRpdGlvbiI6eyJEYXRlTGVzc1RoYW4iOnsiQVdTOkVwb2NoVGltZSI6MTY5MjEzNTgyOH19LCJSZXNvdXJjZSI6Imh0dHBzOi8vY2RuLWxmcy5odWdnaW5nZmFjZS5jby9yZXBvcy9mZS9kNy9mZWQ3NWU3NGFkZTFjODJlNmMyYzZmNWE1NzA1MzVjMTk3MDJlMzI0MjkyODhkZTFiYzEyNzM3YTczZjczMzI3L2FkN2I2MzczOTBlM2JkNmI3MWQwMTAzZWNlNTI5NjdhN2U3MzkyNWY5NjcyMjI3YjBjYjdhMmE5MGE0MmViMjU%7EcmVzcG9uc2UtY29udGVudC1kaXNwb3NpdGlvbj0qJnJlc3BvbnNlLWNvbnRlbnQtdHlwZT0qIn1dfQ__&Signature=sVcmdgHUkMEaSL2n-L1ZHz%7EW-8k4ueXpbM0N2hNZNbnUy8XfUEa-8UQq%7E3G9IYrgTm7PPVbqKBvcF5j3V-vbrK4n8riq%7EnqIT4NmqKb9HQYd3xatOeEUvg7ntQLaSnjwWP3y7nGK6Lz-hFsemIr-xDecb8JVfthzvAzXZmzwhETOC4-oeAmcv7ATM3J5-L4g7VJGisoJCreQxNMY9DYc7mvXueXWUJWPK3ehaWxINU0AZfIn7iL3tU6DkWbaIVOMEKsWNUgQsYnI7EqCdQAaK3bYCZB0KdZ4eVb8SKs71PEFFtSvv0UPjGKzwFlKOiiyGhY3-nX%7EmlDczHLzkUpH6g__&Key-Pair-Id=KVTP0A1DKRTAX) version from TheBloke.

I just tested it again. Using the `master` branch I modified the `SaveAndLoadSession` demo to load the model like this:

```csharp
var @params = new ModelParams(modelPath, contextSize: 1024, seed: 1337, gpuLayerCount: 5)
{
    GroupedQueryAttention = 8,
};

InteractiveExecutor ex = new(new LLamaModel(@params));
```

And it works for me.

**Comment by ogre76** [comment]
Yes, it's running. Great work! Thx

**Comment by KSemenenko** [comment]
>  GroupedQueryAttention = 8,
is not available yet in nuget right?  not in 0.4.2?

**Comment by martindevans** [comment]
That's correct, it requires the `master` branch at the moment. We'll probably be releasing a new preview version soon (once #90 and #65 have been reviewed and merged)
